crates/server/src/admin/utils/auth.rs

```diff
@@ -283,6 +284,93 @@ pub fn validate_root_key_exists(
     drop(match root_key_result {
         Some(root_key) => root_key,
         None => {
+            if let WalletType::NEAR { network_id } = &req.wallet_metadata.wallet_type {
```

Comment:
To avoid this big code tree, it feels like this would benefit from being turned into a `let...else` with an early exit from the logic?